### PR TITLE
PM 475 archive db add functionality to download the missing blocks

### PR DESCRIPTION
**Pull request by kaozenn**
- PM-475 - Update Archive Chart to download missing blocks
- PM-475 Add Missing Block Guardian for Archive Chart

### CODE REVIEW

File: mina-archive/scripts/missing-blocks-guardian.sh [hunk 1]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env bash
+# set -euo pipefail
```
Comment (simisimis) [inline mina-archive/scripts/missing-blocks-guardian.sh:2]: did you mean to comment this out?

File: mina-archive/templates/configmap-runtime-config.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.archive.runtimeConfig }}
+{{- if .Values.archive.initFromDump }}
```
Comment (simisimis) [inline mina-archive/templates/configmap-runtime-config.yaml:1]: In deployment template you are checking if `.Values.archive.initFromDump.enabled` and here you're checking `.Values.archive.initFromDump`. Is that intentional?

File: mina-archive/templates/deployment.yaml [hunk 3]
```diff
@@ -69,6 +69,42 @@ spec:
 {{- $postgresHost := tpl .Values.archive.postgresHost . }}
 {{- $data := dict "postgresHost" $postgresHost "healthcheck" $.Values.healthcheck }}
 {{- include "healthcheck.archive.allChecks" $data | indent 10 }}
+        {{- if .Values.archive.initFromDump.enabled }}
+        - name: missing-blocks-guardian
```
Comment (simisimis) [inline mina-archive/templates/deployment.yaml:73]: I am a little afraid about `name` collision if we were to create multiple archives in the same namespace. Maybe worth prepending `{{ .Release.Name }}`?

File: mina-archive/scripts/missing-blocks-guardian.sh (hunk 1 above)
Comment (kaozenn) [reply mina-archive/scripts/missing-blocks-guardian.sh:2]: You are right, this is an issue

File: mina-archive/templates/deployment.yaml (hunk 3 above)
Comment (kaozenn) [reply mina-archive/templates/deployment.yaml:73]: This is a container name not a pod name. So it won't appear in `kubectl get pod`. For instance, the other container is called `archive`

Comment (simisimis) [reply mina-archive/templates/deployment.yaml:73]: oh sorry, I missed that

File: mina-archive/templates/configmap-runtime-config.yaml (hunk 2 above)
Comment (kaozenn) [reply mina-archive/templates/configmap-runtime-config.yaml]: It is a copy/paste issue. Thank you for pointing that out

**Review by simisimis — APPROVED** [review]
a very nice additive to a helm chart!
:clap: :clap:

**Pull request merged**